src/jit/codegenarmarch.cpp

```diff
@@ -1359,6 +1359,11 @@ void CodeGen::genCodeForIndir(GenTreeIndir* tree)
     genConsumeAddress(tree->Addr());
     emit->emitInsLoadStoreOp(ins_Load(targetType), emitTypeSize(tree), targetReg, tree);
     genProduceReg(tree);
+
+    if (tree->gtFlags & GTF_IND_VOLATILE)
+    {
+        instGen_MemoryBarrier();
+    }
```

Comment:
Also I think that the location of the memory barrier may change if we have a store vs. a load
( figured out after I wrote this comment,  that this  case is always a load and the memory barrier comes after)